src/convert.jl

```diff
@@ -1247,3 +1247,49 @@ function convert(::Type{SimpleSparsePolynomialZonotope}, Z::AbstractZonotope)
 
     return SimpleSparsePolynomialZonotope(c, G, E)
 end
+
+"""
+    convert(::Type{SparsePolynomialZonotope}, Z::Zonotope)
+
+Convert a zonotope to sparse polynomial zonotope.
+"""
+function convert(::Type{SparsePolynomialZonotope}, Z::AbstractZonotope{N}) where {N}
+    c = center(Z)
+    G = genmat(Z)
+    n = ngens(Z)
+    E = Matrix(1 * I, n, n)
+    idx = uniqueID(n)
+    GI = zeros(N, dim(Z), 0)
+
+    return SparsePolynomialZonotope(c, G, GI, E, idx)
+end
+
+"""
+    convert(::Type{SparsePolynomialZonotope}, SSPZ::SimpleSparsePolynomialZonotope)
+
+Convert a simple sparse polynomial zonotope to a sparse polynomial zonotope.
+"""
+function convert(::Type{SparsePolynomialZonotope}, SSPZ::SimpleSparsePolynomialZonotope{N}) where {N}
+    c = center(SSPZ)
+    G = genmat(SSPZ)
+    E = expmat(SSPZ)
+    n = ngens(SSPZ)
+
+    GI = zeros(N, dim(SSPZ), 0)
+
+    return SparsePolynomialZonotope(c, G, GI, E, uniqueID(n))
+end
+
+"""
+    convert(::Type{SimpleSparsePolynomialZonotope}, SPZ::SparsePolynomialZonotope)
+
+Convert a sparse polynomial zonotope to simple sparse polynomial zonotope.
+"""
+function convert(::Type{SimpleSparsePolynomialZonotope}, SPZ::SparsePolynomialZonotope)
+    c = center(SPZ)
+    G = genmat(SPZ)
+    n = nindependentgens(SPZ)
+    E = cat(expmat(SPZ), Matrix(1 * I, n, n), dims=(1, 2))
+
+    return SimpleSparsePolynomialZonotope(c, G, E)
+end
```

Comment:
In what cases wouldn't it work? I guess the empty identity matrix is the only corner case? Will check if it works in that case

Reply:
I mean: You completely ignore the dependent generators. How can this be an exact conversion?

Reply:
no, `genmat` returns the concatenated dependent and independent generators, `E` already contains the exponents of the dependent ones. Since the independent ones are degree 1, their exponent matrix is the identity matrix and the new exponent matrix is the their block diagonal matrix

![image](https://user-images.githubusercontent.com/49938764/182089634-5da453fe-304c-41e8-8b81-40cd007ae98c.png)

Niklas thesis doesn't have "SimpleSparsePolynomialZonotopes", hence it keeps an empty matrix of independent generators and expands the index vector, the simple version ignores those

Reply:
I guess the terminology is a little tricky, in SPZ *dependent* parameters can have higher degree and independent ones are first degree only. In SSPZ you only have independent variables (no need to keep the index vector) but then those can have higher exponent (otherwise would be just a zonotope)

Reply:
Thanks. And we have an application of `genmat` :tada: